File: leetcode/AIcoding/1.Gemini-2.5/go_solutions/645-set-mismatch.go
func findErrorNums(nums []int) []int {
	n := len(nums)
	seen := make([]bool, n+1)

	duplicate := 0
	missing := 0

	for _, num := range nums {
		if seen[num] {
			duplicate = num
		}
		seen[num] = true
	}

	for i := 1; i <= n; i++ {
		if !seen[i] {
			missing = i
			break
		}
	}

	return []int{duplicate, missing}
}

File: leetcode/AIcoding/5.Qwen-3/java_solutions/645-set-mismatch.java
public class Solution {

import java.util.*;

public class Solution {
    public int[] findErrorNums(int[] nums) {
        int n = nums.length;
        int[] result = new int[2];
        Set<Integer> seen = new HashSet<>();

        for (int num : nums) {
            if (seen.contains(num)) {
                result[0] = num;
            } else {
                seen.add(num);
            }
        }

        for (int i = 1; i <= n; i++) {
            if (!seen.contains(i)) {
                result[1] = i;
                break;
            }
        }

        return result;
    }
}
}